stdlib/src/utils/string_slice.mojo

```diff
@@ -1158,7 +1155,32 @@ fn _to_string_list[
 
 
 @always_inline
-fn _to_string_list[
+fn to_string_list[
+    O: MutableOrigin, //
+](items: List[StringSlice[O]]) -> List[String]:
```

Comment:
**Suggestion** I wonder if we could avoid duplicating `to_string_list()` (once for `ImmutableOrigin`, once for `MutableOrigin`) and instead have a single definition with the signature:

```mojo
fn to_string_list[
    mut: Bool,
    origin: Origin[mut], //
](items: List[StringSlice[O]]) -> List[String]:
```

Going a step further, if we rely on autoparameterization, I think we could make the signature even simpler by eliding the parameters, and just have:

```mojo
fn to_string_list(items: List[StringSlice]) -> List[String]:
```

That definition will still accept any immutable or mutable `StringSlice`, but we don't have to spell the parameters out explicitly.

Reply:
> Going a step further, if we rely on autoparameterization, I think we could make the signature even simpler by eliding the parameters, and just have:
> 
> ```mojo
> fn to_string_list(items: List[StringSlice]) -> List[String]:
> ```
> 
> That definition will still accept any immutable or mutable `StringSlice`, but we don't have to spell the parameters out explicitly.

That's part of my wishlist, I've tried it before: `note: failed to infer parameter 'mut', parameter isn't used in any argument
fn to_string_list(items: List[StringSlice]) -> List[String]:`

> **Suggestion** I wonder if we could avoid duplicating `to_string_list()` (once for `ImmutableOrigin`, once for `MutableOrigin`) and instead have a single definition with the signature:
> 
> ```mojo
> fn to_string_list[
>     mut: Bool,
>     origin: Origin[mut], //
> ](items: List[StringSlice[O]]) -> List[String]:
> ```

Totally forgot hehe, worked like a charm